### PR TITLE
added frenchToastAlert Command

### PR DESCRIPTION
**Pull request by garth-vader**

### CODE REVIEW

**Comment by jbpratt** [comment]
Hi @garth-vader , thanks for the PR.

Can you explain the purpose of this command?

File: commands.go [hunk 1]
```diff
@@ -114,6 +117,31 @@ func (b *bot) sudoku(m dggchat.Message, s *dggchat.Session) {
 	s.SendMute(m.Sender.Nick, -1)
 }
 
+func (b *bot) frenchToastAlert(m dggchat.Message, s *dggchat.Session) {
+	type ftlXML struct {
+		XMLName xml.Name `xml:"frenchtoast"`
+		Status  string   `xml:"status"`
+	}
```
Comment (jbpratt) [inline commands.go:124]:
```suggestion
	f := struct {
		XMLName xml.Name `xml:"frenchtoast"`
		Status  string   `xml:"status"`
	}{}
```

File: commands.go [hunk 2]
```diff
@@ -114,6 +117,31 @@ func (b *bot) sudoku(m dggchat.Message, s *dggchat.Session) {
...
+	if err != nil {
+		return
+	}
+	//Parse Alert
+	if resp.StatusCode != 200 {
+		b.sendMessageDedupe("Error reaching French Toast Alert", s)
+		return
+	}
+	byteValue, _ := ioutil.ReadAll(resp.Body)
+	var f ftlXML
```
Comment (jbpratt) [inline commands.go:139]:
```suggestion
```

File: commands.go [hunk 3]
```diff
@@ -114,6 +117,31 @@ func (b *bot) sudoku(m dggchat.Message, s *dggchat.Session) {
...
+	xml.Unmarshal(byteValue, &f)
```
Comment (jbpratt) [inline commands.go:140]: Let's check the error to ensure unmarshalling succeeded, if not then we would log out a message and return early.

File: commands.go [hunk 4]
```diff
@@ -114,6 +117,31 @@ func (b *bot) sudoku(m dggchat.Message, s *dggchat.Session) {
...
+	//Bot says Alert
+	b.sendMessageDedupe("Current FT Level is "+f.Status, s)
```
Comment (jbpratt) [inline commands.go:142]:
```suggestion
	b.sendMessageDedupe(fmt.Sprintf("Current FT Level is %s", f.Status), s)
```

File: commands.go [hunk 5]
```diff
@@ -114,6 +117,31 @@ func (b *bot) sudoku(m dggchat.Message, s *dggchat.Session) {
 	s.SendMute(m.Sender.Nick, -1)
 }
 
+func (b *bot) frenchToastAlert(m dggchat.Message, s *dggchat.Session) {
+	type ftlXML struct {
+		XMLName xml.Name `xml:"frenchtoast"`
+		Status  string   `xml:"status"`
+	}
+	if !strings.HasPrefix(m.Message, "!frenchToastAlert") {
```
Comment (jbpratt) [inline commands.go:125]: If we were to accept this (still undecided) I think I would prefer `!frenchtoast`

File: commands.go [hunk 6]
```diff
@@ -114,6 +117,31 @@ func (b *bot) sudoku(m dggchat.Message, s *dggchat.Session) {
...
+	resp, err := http.Get("https://www.universalhub.com/toast.xml")
+	if err != nil {
+		return
+	}
+	//Parse Alert
+	if resp.StatusCode != 200 {
+		b.sendMessageDedupe("Error reaching French Toast Alert", s)
+		return
+	}
+	byteValue, _ := ioutil.ReadAll(resp.Body)
```
Comment (jbpratt) [inline commands.go:138]: Let's check the error here.

**Review by jbpratt — CHANGES_REQUESTED** [review]
Left some comments on small improvements. One of my big hold-ups on this is how many people are actually being impacted by the same weather at any given time.

If you wanted to take this further, I would probably recommend developing a stand-alone `weather` bot, allowing it to take things like `location` and return back a weather response (http://frenchtoastalert.com/)

File: commands.go (hunk 1 above)
Comment (garth-vader) [reply commands.go:124]: Yeah I only made this as a meme for like 2 chatters (it is a regional new england thing). 
But I could look into making it into a command to see if you should go out and get groceries, before a large amount of snow is about to hit you.

**Comment by jbpratt** [comment]
closing this per discussion in chat, will be moved to @SoMuchForSubtlety 's weather bot if possible. Thank you!

**Pull request closed without merging**